src/snyk/snykCode/codeService.ts

```diff
@@ -198,18 +198,18 @@ export class SnykCodeService extends AnalysisStatusProvider implements ISnykCode
       }
     } catch (err) {
       this.temporaryFailed = true;
-      await this.errorHandler.processError(err, undefined, () => {
+      await this.errorHandler.processError(err, { requestId }, () => {
         this.errorEncountered(requestId);
       });
 
-      if (enabledFeatures?.codeSecurityEnabled) {
+      if (enabledFeatures?.codeSecurityEnabled && this.errorHandler.transientErrorsLimitExhausted) {
```

Comment:
so we log it if `transientErrorsLimitExhausted` true? Either that is misnamed or I don't understand the logic.

Reply:
so the `transientErrorsLimitExhausted` correlates to the logic [here](https://github.com/snyk/vscode-extension/blob/d31f7d09d57c6b6efc6b79687a2642e65aa01139/src/snyk/snykCode/error/snykCodeErrorHandler.ts#L120). So essentially we retry failed request a bunch of times and check whether the amount of error'd tries are less than `MAX_CONNECTION_RETRIES`. Once that limit is exhausted or hit only then we should be logging an error event to amplitude since that will impact our customer's workflow. 

Reason for the naming was to match the naming of [`transientErrors`](https://github.com/snyk/vscode-extension/blob/d31f7d09d57c6b6efc6b79687a2642e65aa01139/src/snyk/snykCode/error/snykCodeErrorHandler.ts#L21) but happy to change it to `connectionRetryLimitExhausted` or something along those lines for better readability?

Reply:
Let's go with your connectionRetry suggestion.